File: backend/src/main/java/org/icpclive/events/codeforces/api/data/CFProblem.java
package org.icpclive.events.codeforces.api.data;

import java.util.List;

/**
 * @author egor@egork.net
 */
public class CFProblem {
    public Integer contestId;
    public String problemsetName;
    public String index;
    public String name;
    public CFProblemType type;
    public Double points;
    public Integer rating;
    public List<String> tags;

    public enum CFProblemType {
        PROGRAMMING,
        QUESTION
    }
}

File: backend/src/main/kotlin/org/icpclive/cds/codeforces/api/data/CFProblem.kt
@file:Suppress("UNUSED")
package org.icpclive.cds.codeforces.api.data

import kotlinx.serialization.*

enum class CFProblemType {
    PROGRAMMING, QUESTION
}


@Serializable
data class CFProblem(
    val contestId: Int,
    val problemsetName: String? = null,
    val index: String,
    val name: String? = null,
    val type: CFProblemType,
    val points: Double? = null,
    val rating: Int? = null,
    val tags: List<String>? = null,
)